### ISSUE TITLE
How to get the token probability by curl request?

### ISSUE THREAD

**Issue by TWSFar** [huggingface/text-generation-inference#2452]
### Feature request

curl -v -X POST http://.....srv/generate -H "Content-Type: application/json"  -d '{"inputs": "xxxxx:","parameters": {"max_new_tokens": 256}}'
user this curl request, get output like
{"generated_text": xxxx}

how to get generated text probability from llm in TGI service?

### Motivation

no

### Your contribution

no

**Comment by ErikKaum** [comment]
Hi @TWSFar 👋 

Thanks for the question, for example with the command:

```bash
curl 127.0.0.1:3000/generate_stream \
    -X POST \
    -d '{"inputs":"What is Deep Learning?","parameters":{"max_new_tokens":20}}' \
    -H 'Content-Type: application/json'
```

the output has the log probabilities:
```
data: {"index":1,"token":{"id":89554,"text":" Deep","logprob":-0.97209036,"special":false},"generated_text":null,"details":null}

data: {"index":2,"token":{"id":89950,"text":" Learning","logprob":-0.3902701,"special":false},"generated_text":null,"details":null}

data: {"index":3,"token":{"id":632,"text":" is","logprob":-0.568623,"special":false},"generated_text":null,"details":null}

data: {"index":4,"token":{"id":267,"text":" a","logprob":-0.7765734,"special":false},"generated_text":null,"details":null}

data: {"index":5,"token":{"id":2084,"text":" new","logprob":-2.356847,"special":false},"generated_text":null,"details":null}
```

**Comment by TWSFar** [comment]
> log probabilities:

logprob Is logits (the original output of LLM) or the log of the probability value (the original output is first softmaxed and then log)?

**Comment by ErikKaum** [comment]
The "logprob" values you're are the log of the probability values, not the raw logits 👍
